src/cmake/build_gif.cmake

```diff
@@ -0,0 +1,43 @@
+# Copyright Contributors to the OpenImageIO project.
+# SPDX-License-Identifier: Apache-2.0
+# https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+######################################################################
+# GIF by hand!
+######################################################################
+
+set_cache (GIF_BUILD_VERSION 5.2.1 "GIFLIB version for local builds")
+set (GIF_URL "https://downloads.sourceforge.net/project/giflib/giflib-${GIF_BUILD_VERSION}.tar.gz")
+set (GIF_URL_HASH SHA256=31da5562f44c5f15d63340a09a4fd62b48c45620cd302f77a6d9acf0077879bd)
+
+set_cache (GIF_BUILD_SHARED_LIBS ${LOCAL_BUILD_SHARED_LIBS_DEFAULT}
+           DOC "Should execute a local GIF build, if necessary, build shared libraries" ADVANCED)
+
+string (MAKE_C_IDENTIFIER ${GIF_BUILD_VERSION} GIF_VERSION_IDENT)
+
+# Build setup using ExternalProject
+build_dependency_with_cmake(GIF
+    VERSION         ${GIF_BUILD_VERSION}
+    URL             ${GIF_URL}
+    URL_HASH        ${GIF_URL_HASH}
+    CMAKE_ARGS
+        -D CMAKE_INSTALL_PREFIX=${GIF_LOCAL_INSTALL_DIR}
+        -D BUILD_SHARED_LIBS=${GIF_BUILD_SHARED_LIBS}
+        -D CMAKE_INSTALL_LIBDIR=lib
+    PATCH_COMMAND
+        curl -L https://sourceforge.net/p/giflib/bugs/_discuss/thread/4e811ad29b/c323/attachment/Makefile.patch -o Makefile.patch &&
+        patch -p0 < Makefile.patch
```

Comment:
Should we be bothered by the fact that the `build_dependency_with_cmake` function doesn't take a "PATCH_COMMAND" argument, and so this extra patch won't be applied?

Reply:
Also, build_dependency_with_cmake doesn't take a URL or URL_HASH argument, and in fact the only kind of download it knows how to do is to clone a git repo? I just don't understand how the code as presented here can work. 

Are you sure you aren't locally running and testing a version of this patch that is different from the one currently in this PR?

This hypothesis is also supported by the fact that the screenshots you attached have messages about FetchContent, which we do not call anywhere and also aren't mentioned in this patch.

Reply:
just realized I forgot to push my latest change where I removed the build_dependency_with_cmake method since it only accepts a git repository